.github/actions/spell-check/whitelist.txt

```diff
@@ -0,0 +1,3240 @@
+rdl
+RDONLY
+rdpartysource
+readback
+READCONSOLE
+READCONSOLEOUTPUT
+READCONSOLEOUTPUTSTRING
+Readline
+readme
+READMODE
+readonly
+READWRITE
+realloc
+reamapping
+rects
+redef
+Redir
+redirector
+redist
+redistributable
+REDSCROLL
+refactor
+refactoring
+REFCLSID
+refcount
+referencesource
+REFGUID
+REFIID
+REFPROPERTYKEY
+regex
+REGISTEROS
+REGISTERVDM
+regkey
+REGSTR
+reingest
+Relayout
+RELBINPATH
+renderengine
+rendersize
+reparenting
+replatformed
+Replymessage
+repositorypath
+rerendered
+rescap
+Resequence
+Reserialize
+RESETCONTENT
+resheader
+resizable
+resmimetype
+restrictedcapabilities
+resw
+resx
+retval
+rfa
+rfc
+rftp
+rgb
+rgba
+rgbi
+rgch
+rgci
+rgdx
+rgfae
+rgfte
+rgi
+rgn
+rgp
+rgpwsz
+rgrc
+rgs
+rgui
+rgus
+rgw
+rgwch
+rhs
+ri
+richturn
+RIGHTALIGN
+RIGHTBUTTON
+riid
+Rincewind
+RIPMSG
+Rl
+rmdir
+RMENU
+roadmap
+robomac
+roundtrip
+ROWSTOSCROLL
+rprichard
+Rr
+RRF
+rrggbb
+RRRRRRR
+rsp
+rtcore
+RTEXT
+rtf
+RTFTo
+Rtl
+RTLREADING
+RTTI
+ru
+ruleset
+runas
+runasradio
+RUNDLL
+runformat
+runft
+RUNFULLSCREEN
+runsettings
+runtimeclass
+runuia
+runut
+rvalue
+RVERTICAL
+RWIN
+rying
+safearray
+SAFECAST
+safemath
+sba
+SBCS
+SBCSDBCS
+sbi
+sbiex
+sbold
+sbr
+scancode
+scanline
+scc
+schemaview
+schemename
+Schoonover
+SCL
+scm
+scoll
+scollbars
+scprintf
+SCRBUF
+SCRBUFSIZE
+screenbuffer
+SCREENBUFFERINFO
+screeninfo
+screenshot
+scriptload
+Scro
+Scrollable
+scrollback
+scrollbar
+Scrolldown
+Scrolldownpage
+Scroller
+SCROLLFORWARD
+SCROLLINFO
+scrolllock
+SCROLLSCALE
+SCROLLSCREENBUFFER
+Scrollup
+Scrolluppage
+scursor
+sddl
+sdeleted
+sdf
+sdk
+SDKDDK
+sdktools
+searchbox
+securityappcontainer
+segfault
+Segoe
+SELCHANGE
+SELECTALL
+selectany
+SELECTEDFONT
+SELECTSTRING
+Selfhosters
+SERVERDLL
+Ses
+SETACTIVE
+SETBUDDYINT
+SETCOLOR
+setcp
+SETCURSEL
+SETCURSOR
+SETCURSORINFO
+SETCURSORPOSITION
+SETDISPLAYMODE
+setfill
+SETFOCUS
+SETFONT
+SETFOREGROUND
+SETHARDWARESTATE
+SETHOTKEY
+SETICON
+setintegritylevel
+SETITEMDATA
+SETITEMHEIGHT
+SETKEYSHORTCUTS
+setlocal
+setlocale
+SETMENUCLOSE
+setmode
+SETNUMBEROFCOMMANDS
+SETOS
+SETPALETTE
+SETPOS
+SETRANGE
+SETSCREENBUFFERSIZE
+SETSEL
+SETTEXTATTRIBUTE
+SETTINGCHANGE
+SETTITLE
+setw
+Setwindow
+SETWINDOWINFO
+sfi
+SFUI
+sgr
+SGRXY
+SHCo
+shcore
+shellapi
+shellex
+shellexecute
+shellscalingapi
+SHFILEINFO
+SHGFI
+SHIFTJIS
+Shl
+shlguid
+shlobj
+shlwapi
+SHORTPATH
+SHOWCURSOR
+SHOWMAXIMIZED
+SHOWMINNOACTIVE
+SHOWNOACTIVATE
+SHOWNORMAL
+SHOWWINDOW
+SHRT
+sidebyside
+SIF
+SIGDN
+Silverlight
+SINGLEFLAG
+SINGLETHREADED
+siup
+sizeof
+SIZESCROLL
+SKIPFONT
+SKIPOWNPROCESS
+SKIPOWNTHREAD
+sku
+sldl
+SLGP
+SLIST
+slmult
+sln
+slpit
+smalllogo
+SManifest
+SMARTQUOTE
+SMTO
+SND
+SOLIDBOX
+Solutiondir
+somefile
+Somuah
+sonph
+sonpham
+sourced
+spammy
+spand
+Spc
+spkg
+splashscreen
+sprintf
+sqlproj
+srand
+src
+SRCCODEPAGE
+SRCCOPY
+SRCINVERT
+srcsrv
+SRCSRVTRG
+srctool
+sre
+srect
+srv
+srvinit
+srvpipe
+ssangtikeut
+ssh
+sstream
+stackoverflow
+stakx
+standalone
+STARTF
+STARTUPINFO
```

Comment:
I wonder if I can give you a path to a header file and have you presume the spelling of anything in that file is a valid word?

Reply:
Hmm.. That's definitely doable.
That isn't precisely how the current dictionary is loaded, but it's close enough.

I'll chew on this.

Would you want one file or a bunch?

Reply:
A bunch. I'm just thinking that we could give it a couple of Windows headers, ATL headers, STL headers, and CRT headers that are representative of containing a lot of the structs/words we use. Could read them ahead of time and make sure the UPSTREAM folks didn't typo them and then make them effectively "part of the dictionary". I dunno, not every idea I has is necessary good in practice.

Reply:
Ok. Would you prefer to feed it raw source files or whitelists generated from them? (Assuming we're using directories)

My preference is the latter.

And, as you've noticed, upstreams do not tend to be any better than downstream. I'm constantly running across errors that trickled around.

Part of my goals in this are to prevent the next HTTP `Referrer`.

Reply:
Reading this, it seems like we're really treating these as dictionary entries as opposed to whitelist entries -- https://github.com/microsoft/terminal/pull/4799#discussion_r389340977.

Reply:
You're the one immersed in this world, so if you say a list generated from them is better, I'm inclined to do that. If we could generate a few "dictionaries" for things with a utility and check those in, I'd be happy with that.